support/itemUtils.ts

```diff
@@ -106,39 +100,38 @@ export function createMapFromItem(
 
   return isWebMap
     ? createWebMapFromItem(options)
-    : (createWebSceneFromItem(options) as IPromise<WebMap | WebScene>);
+    : (createWebSceneFromItem(options) as Promise<esri.WebMap | esri.WebScene>);
 }
 
-export function createWebMapFromItem(
+export async function createWebMapFromItem(
   options: CreateMapFromItemOptions
-): IPromise<WebMap> {
+): Promise<esri.WebMap> {
   const { item, appProxies } = options;
-  return requireUtils.when(require, "esri/WebMap").then(WebMap => {
-    const wm = new WebMap({
-      portalItem: item
-    });
-    return wm.load().then(() => {
-      return wm.basemap.load().then(() => {
-        return _updateProxiedLayers(wm, appProxies);
-      });
+  const WebMap = await import("esri/WebMap");
+  const wm = new WebMap.default({
+    portalItem: item
+  });
+  return wm.load().then(() => {
```

Comment:
Can we use async await for these? We shouldn't have to use any more `then()`

Reply:
Yeah... that's part of the work I want to do next. I just wanted to get something we could use in the web pack sample @rslibed  is working on.

Reply:
I think we discussed that I'm going to implement async/await. I'll get on that asap.

Reply:
This one is done.